test/shared/Maybe.js

```diff
@@ -43,6 +43,11 @@ Maybe.prototype['fantasy-land/ap'] = function(maybe) {
   return this.isJust && maybe.isJust ? Just(maybe.value(this.value)) : Nothing;
 };
 
+//  Maybe#promap :: Maybe (a -> b) ~> (c -> a, b -> d) -> Maybe (c -> d)
```

Comment:
`Maybe a` cannot satisfy the requirements of [Profunctor][1] for two reasons, the simpler of which is that `Maybe a` has only one type variable.

Let's see why `Function a b` *can* satisfy the requirements of Profunctor, starting from the generalized `fantasy-land/promap` signature:

```haskell
fantasy-land/promap :: Profunctor p => p b c ~> (a -> b, c -> d) -> p a d
```

Let's replace `Profunctor p => p` with `Function`:

```haskell
fantasy-land/promap :: Function b c ~> (a -> b, c -> d) -> Function a d
```

Finally, let's use the more familiar `(a -> b)` syntax:

```haskell
fantasy-land/promap :: (b -> c) ~> (a -> b, c -> d) -> (a -> d)
```

Given a function from `a` to `b` and a function from `c` to `d`, can we transform `Function b c` into `Function a d`? Yes. We can return a function that receives an `a`, we can get a `b` from the `a`, we can then get a `c` from the `b`, and we can then get a `d` from the `c`.

Let's replace `Profunctor p => p` with `Pair`:

```haskell
fantasy-land/promap :: Pair b c ~> (a -> b, c -> d) -> Pair a d
```

This is impossible, as we do not have an `a` and we have no way to get an `a` from a `b` or a `c`. For this reason `Pair a b` cannot satisfy the requirements of Profunctor.


[1]: https://github.com/fantasyland/fantasy-land#profunctor

Reply:
I see, so any Profunctor instance needs to be parameterized as `p a b`; I can't fudge it by providing something else parameterized by `a b` in the instances `a` slot, i.e. `Identity (a -> b)`. I was thinking it might work something like how `Monad a` works with functions by partially applying the input param of `(->)`: `Monad ((->) r)`. I suppose this works because it's still `Monad a`, while `Identity (a -> b)` is not `Profunctor a b`. I'll remove these and use a valid Profunctor instance!

Reply:
I suppose `Pair (-> a) (a ->)` could work?

Reply:
indeed - `Pair` has a `Profunctor` instance for `(Profunctor p, Profunctor q) => Pair p q`. PR updated 👍